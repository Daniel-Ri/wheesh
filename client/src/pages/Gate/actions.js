import {
  GET_ALL_STATIONS,
  SET_STATIONS,
  SET_VALIDATE_RESULT,
  VALIDATE_TICKET_ON_ARRIVAL,
  VALIDATE_TICKET_ON_DEPARTURE,
} from './constants';

export const getAllStations = () => ({
  type: GET_ALL_STATIONS,
});

export const setStations = (stations) => ({
  type: SET_STATIONS,
  stations,
});

export const validateTicketOnDeparture = (inputs, handleSuccess, handleError) => ({
  type: VALIDATE_TICKET_ON_DEPARTURE,
  inputs,
  handleSuccess,
  handleError,
});

export const validateTicketOnArrival = (inputs, handleSuccess, handleError) => ({
  type: VALIDATE_TICKET_ON_ARRIVAL,
  inputs,
  handleSuccess,
  handleError,
});

export const setValidateResult = (validateResult) => ({
  type: SET_VALIDATE_RESULT,
  validateResult,
});
