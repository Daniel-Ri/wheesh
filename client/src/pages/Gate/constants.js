export const GET_ALL_STATIONS = 'Gate/GET_ALL_STATIONS';
export const SET_STATIONS = 'Gate/SET_STATIONS';
export const VALIDATE_TICKET_ON_DEPARTURE = 'Gate/VALIDATE_TICKET_ON_DEPARTURE';
export const VALIDATE_TICKET_ON_ARRIVAL = 'Gate/VALIDATE_TICKET_ON_ARRIVAL';
export const SET_VALIDATE_RESULT = 'Gate/SET_VALIDATE_RESULT';
