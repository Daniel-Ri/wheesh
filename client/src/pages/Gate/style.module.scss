@import '@styles/base';

// Reset
* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: Poppins;
}

.main {
  flex-grow: 1;
  margin-left: 200px;
  position: relative;
  background: $color-alabaster;
  display: flex;
  justify-content: center;
  width: calc(100vw - 200px);

  .container {
    width: 90%;
    max-width: 40rem;
    @include directionY;

    header {
      width: 100%;
      margin-top: 5vh;
      @include directionX;
      justify-content: center;
      align-items: center;
      position: relative;

      h1 {
        font-size: 120%;
        width: 60%;
        text-align: center;
      }
    }

    section {
      margin-top: 1rem;
      background-color: $color-white;
      padding: 1.5rem 1rem;
      border-radius: 10px;
      box-shadow: 0 0 5px 0 $color-cool-gray;
      @include directionY;
      gap: 1rem;

      .row {
        width: 100%;
        @include directionX;
        align-items: center;
        justify-content: space-between;
        gap: 0.75rem;

        label {
          font-size: 80%;
          color: $color-cool-gray;
        }

        .input {
          width: 45%;
          @include directionY;
          
        }
      }

      .rowReader {
        width: 100%;
        @include directionX;
        justify-content: center;
      }
    }

    .result {
      @include directionY;
      gap: 0.5rem;
      align-items: center;
    }
  }

  @media screen and (max-width: $screen-lg-limit) {
    margin-left: 192px;
    width: calc(100vw - 192px);
  }

  @media screen and (max-width: $screen-md-limit) {
    margin-left: 10rem;
    width: calc(100vw - 10rem);
  }

  @media screen and (max-width: $screen-sm-limit) {
    margin-left: 0;
    width: 100vw;

    .container {
      header {
        height: 3rem;
      }
    }
  }
}