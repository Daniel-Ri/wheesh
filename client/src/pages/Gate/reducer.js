import { produce } from 'immer';
import { SET_STATIONS, SET_VALIDATE_RESULT } from './constants';

export const initialState = {
  stations: [],
  validateResult: null,
};

export const storedKey = [];

const gateReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_STATIONS:
        draft.stations = action.stations;
        break;
      case SET_VALIDATE_RESULT:
        draft.validateResult = action.validateResult;
        break;
    }
  });

export default gateReducer;
