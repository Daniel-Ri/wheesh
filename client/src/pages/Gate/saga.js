import { getAllStations, validateTicketOnArrival, validateTicketOnDeparture } from '@domain/api';
import { takeLatest, call, put } from 'redux-saga/effects';
import { GET_ALL_STATIONS, VALIDATE_TICKET_ON_ARRIVAL, VALIDATE_TICKET_ON_DEPARTURE } from './constants';
import { setStations, setValidateResult } from './actions';

function* doGetAllStations() {
  try {
    const response = yield call(getAllStations);
    yield put(setStations(response.data));
  } catch {
    // error
  }
}

function* doValidateTicketOnDeparture({ inputs, handleSuccess, handleError }) {
  try {
    const response = yield call(validateTicketOnDeparture, inputs);
    yield put(setValidateResult(response.data));
    yield call(handleSuccess);
  } catch (error) {
    handleError(error.response.data.message);
  }
}

function* doValidateTicketOnArrival({ inputs, handleSuccess, handleError }) {
  try {
    const response = yield call(validateTicketOnArrival, inputs);
    yield put(setValidateResult(response.data));
    yield call(handleSuccess);
  } catch (error) {
    handleError(error.response.data.message);
  }
}

export default function* gateSaga() {
  yield takeLatest(GET_ALL_STATIONS, doGetAllStations);
  yield takeLatest(VALIDATE_TICKET_ON_DEPARTURE, doValidateTicketOnDeparture);
  yield takeLatest(VALIDATE_TICKET_ON_ARRIVAL, doValidateTicketOnArrival);
}
