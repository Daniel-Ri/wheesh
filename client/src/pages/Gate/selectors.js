import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectGateState = (state) => state.gate || initialState;

export const selectStations = createSelector(selectGateState, (state) => state.stations);
export const selectValidateResult = createSelector(selectGateState, (state) => state.validateResult);
