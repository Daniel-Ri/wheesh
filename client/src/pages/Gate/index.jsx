/* eslint-disable jsx-a11y/label-has-associated-control */
import { useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import { connect, useDispatch } from 'react-redux';
import { Html5QrcodeScanner } from 'html5-qrcode';

import { createStructuredSelector } from 'reselect';
import { Autocomplete, Button, FormControl, MenuItem, Select, TextField } from '@mui/material';
import toast from 'react-hot-toast';
import classes from './style.module.scss';
import { getAllStations, setValidateResult, validateTicketOnArrival, validateTicketOnDeparture } from './actions';
import { selectStations, selectValidateResult } from './selectors';

const Gate = ({ stations, validateResult }) => {
  const dispatch = useDispatch();

  const [direction, setDirection] = useState('depart');
  const [station, setStation] = useState(null);
  const [isScan, setIsScan] = useState(false);
  const [isScannerRendering, setIsScannerRendering] = useState(false);
  const [scanResult, setScanResult] = useState(null);

  // Use a ref for isScan
  const isScanRef = useRef(isScan);
  isScanRef.current = isScan;

  const defaultProps = {
    options: stations,
    getOptionLabel: (option) => option.name,
  };

  const handleSuccessValidate = () => {
    toast.success('Success validate QR');
  };

  const handleErrorValidate = (errorMsg) => {
    toast.error(errorMsg);
  };

  useEffect(() => {
    dispatch(getAllStations());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    let scanner;

    const clearScanner = () => {
      if (scanner) {
        scanner.clear();
        setIsScannerRendering(false);
        setIsScan(false);
      }
    };

    if (isScan) {
      if (isScannerRendering) return;

      scanner = new Html5QrcodeScanner('reader', {
        qrbox: {
          width: 450,
          height: 450,
        },
        fps: 5,
        rememberLastUsedCamera: true,
      });

      const handleSuccess = (result) => {
        clearScanner();
        dispatch(setValidateResult(null));
        setScanResult(result);
      };

      const handleError = (error) => {
        console.warn(error);

        if (!isScanRef.current) {
          clearScanner();
        }
      };

      scanner.render(handleSuccess, handleError);
      setIsScannerRendering(true);
    } else {
      clearScanner();
    }

    return () => {
      clearScanner();
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isScan]);

  useEffect(() => {
    if (direction && station) {
      setIsScan(true);
    } else {
      setIsScan(false);
    }
    setScanResult(null);
    dispatch(setValidateResult(null));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [direction, station, setDirection, setStation]);

  useEffect(() => {
    if (isScan || !scanResult || !station) return;

    let jsonScan;
    try {
      jsonScan = JSON.parse(scanResult);
      if (!jsonScan.id || !jsonScan.secret) {
        throw Error('Invalid QR Ticket');
      }
    } catch {
      toast.error('Invalid QR Ticket');
      return;
    }

    const formattedInputs = {
      orderedSeatId: jsonScan.id,
      secret: jsonScan.secret,
    };
    if (direction === 'depart') {
      formattedInputs.departureStationId = station.id;
      dispatch(validateTicketOnDeparture(formattedInputs, handleSuccessValidate, handleErrorValidate));
    } else {
      formattedInputs.arrivalStationId = station.id;
      dispatch(validateTicketOnArrival(formattedInputs, handleSuccessValidate, handleErrorValidate));
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isScan, scanResult]);

  return (
    <main className={classes.main}>
      <div className={classes.container}>
        <header>
          <h1>Gate</h1>
        </header>
        <section>
          <div className={classes.row}>
            <div className={classes.input}>
              <label>Direction</label>
              <FormControl variant="standard">
                <Select
                  labelId="demo-simple-select-label"
                  id="demo-simple-select"
                  value={direction}
                  onChange={(e) => setDirection(e.target.value)}
                >
                  <MenuItem value="depart">Departure</MenuItem>
                  <MenuItem value="arrive">Arrival</MenuItem>
                </Select>
              </FormControl>
            </div>
            <div className={classes.input}>
              <label>Station</label>
              <Autocomplete
                {...defaultProps}
                id="controlled-demo"
                value={station}
                onChange={(event, newValue) => {
                  setStation(newValue);
                }}
                renderInput={(params) => <TextField {...params} placeholder="Select station" variant="standard" />}
              />
            </div>
          </div>

          {isScan && (
            <div className={classes.rowReader}>
              <div id="reader" />
            </div>
          )}
        </section>

        {!isScan && station && scanResult && (
          <div className={classes.result}>
            {validateResult && (
              <div className={classes.validate}>
                {direction === 'depart' ? (
                  <div className={classes.message}>
                    Have a nice journey to {validateResult.Order.Schedule.arrivalStation.name}, {validateResult.name}
                  </div>
                ) : (
                  <div className={classes.message}>
                    Hope you had a safe journey from {validateResult.Order.Schedule.departureStation.name},{' '}
                    {validateResult.name}
                  </div>
                )}
              </div>
            )}
            <Button variant="contained" onClick={() => setIsScan(true)}>
              Scan Again
            </Button>
          </div>
        )}
      </div>
    </main>
  );
};

Gate.propTypes = {
  stations: PropTypes.array,
  validateResult: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  stations: selectStations,
  validateResult: selectValidateResult,
});

export default connect(mapStateToProps)(Gate);
