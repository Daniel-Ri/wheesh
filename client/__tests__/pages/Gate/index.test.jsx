import { render } from '@utils/testHelper';
import Gate from '@pages/Gate';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

let wrapper;

const mockProps = {
  stations: [
    {
      id: 1,
      name: 'Halim',
    },
    {
      id: 2,
      name: 'Karawang',
    },
    {
      id: 3,
      name: 'Padalarang',
    },
    {
      id: 4,
      name: 'Tegalluar',
    },
  ],
  validateResult: {
    id: 1727,
    orderId: 11,
    seatId: 2457,
    price: 600000,
    gender: 'Male',
    dateOfBirth: '1995-11-11T17:00:00.000Z',
    idCard: '1234567890123456',
    name: 'John Doe',
    email: 'johndoe@gmail.com',
    Order: {
      id: 11,
      userId: 2,
      scheduleId: 65,
      isNotified: true,
      Schedule: {
        id: 65,
        trainId: 5,
        departureStationId: 1,
        arrivalStationId: 3,
        departureTime: '2023-12-17T01:45:00.000Z',
        arrivalTime: '2023-12-17T02:15:00.000Z',
        departureStation: {
          id: 1,
          name: 'Halim',
        },
        arrivalStation: {
          id: 3,
          name: 'Padalarang',
        },
      },
    },
  },
};

beforeEach(() => {
  wrapper = render(<Gate {...mockProps} />);
});

describe('Page Gate', () => {
  test('Renders Gate page', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('Gate')).toBeInTheDocument();
  });

  test('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
