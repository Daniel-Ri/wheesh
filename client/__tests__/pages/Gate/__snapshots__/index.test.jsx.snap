// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Gate Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main"
        data-testid="Gate"
      >
        <div
          class="container"
        >
          <header>
            <h1>
              Gate
            </h1>
          </header>
          <section>
            <div
              class="row"
            >
              <div
                class="input"
              >
                <label>
                  Arah
                </label>
                <div
                  class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                >
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-m5hdmq-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby="demo-simple-select-label demo-simple-select"
                      class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                      id="demo-simple-select"
                      role="button"
                      tabindex="0"
                    >
                      Keberangkatan
                    </div>
                    <input
                      aria-hidden="true"
                      aria-invalid="false"
                      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      tabindex="-1"
                      value="depart"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="input"
              >
                <label>
                  Stasiun
                </label>
                <div
                  class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1h51icj-MuiAutocomplete-root"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-l4u8b9-MuiInputBase-root-MuiInput-root"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-invalid="false"
                        autocapitalize="none"
                        autocomplete="off"
                        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1x51dt5-MuiInputBase-input-MuiInput-input"
                        id="controlled-demo"
                        placeholder="Pilih stasiun"
                        role="combobox"
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                      >
                        <button
                          aria-label="Open"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                          tabindex="-1"
                          title="Open"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="ArrowDropDownIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main"
      data-testid="Gate"
    >
      <div
        class="container"
      >
        <header>
          <h1>
            Gate
          </h1>
        </header>
        <section>
          <div
            class="row"
          >
            <div
              class="input"
            >
              <label>
                Arah
              </label>
              <div
                class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
              >
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-m5hdmq-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="demo-simple-select-label demo-simple-select"
                    class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                    id="demo-simple-select"
                    role="button"
                    tabindex="0"
                  >
                    Keberangkatan
                  </div>
                  <input
                    aria-hidden="true"
                    aria-invalid="false"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value="depart"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="input"
            >
              <label>
                Stasiun
              </label>
              <div
                class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1h51icj-MuiAutocomplete-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-l4u8b9-MuiInputBase-root-MuiInput-root"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1x51dt5-MuiInputBase-input-MuiInput-input"
                      id="controlled-demo"
                      placeholder="Pilih stasiun"
                      role="combobox"
                      spellcheck="false"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="Open"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="Open"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ArrowDropDownIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
